lib/faraday/request/retry.rb

```diff
@@ -20,6 +20,7 @@ module Faraday
   #
   class Request::Retry < Faraday::Middleware
 
+    DEFAULT_EXCEPTIONS = [Errno::ETIMEDOUT, 'Timeout::Error', Error::TimeoutError, Faraday::Error::RetriableResponse]
```

Comment:
One potential addition is calling `.freeze` on this array, as exposing this list rather than constructing it on each call to `exceptions` means it's now modifiable by users. That may be okay though.

Reply:
Good point @krmannix.
However being this a constant, I would rather `.freeze` it and make it immutable as you suggested.
If we want to improve the way you simply add new exceptions to the list, we should probably revisit the API and provide a better way to do it

Reply:
great! I'll update. Thanks for the quick response!